### PR TITLE
[WiP] [DNM] [Gone Rogue] [Need more suggestions] chem heating part 2

### PR DESCRIPTION
**Pull request by MadmanMartian**
better bunsen burner sprite. Courtesy of an anonymous threadposter(?)

bunsen burner

more heating/cooling recipes, such as ice to water, and water to ice. More sensible suggestions requested and very much liked.

Fridges (such as the ones the chef has) now actually cool things down that are inside them.

Bunsen burner now has a fuel slot that you have to open to put fuel in, otherwise it won't let you put a beaker on.

Bunsen burner examine is now a lot more informative
- Tells you what fuels it has, if any
- Tells you what's on it, and what it is.
- Tells you if the fuel port is open or not.

Fixed oversight where ghosts could steal your beaker to the nether realm

Fixed discrepancy where people thought the metals you get from grinding the bars up (or pressing the iron button on the chem dispenser) were their molten equivalents.

molten_iron_shitposting

**current bugs**
- Aluminium has disappeared from the chem dispensers, its option is now empty text, but clicking it creates aluminium still

### CODE REVIEW

**Comment by DrSnips** [comment]
Requesting that the recipe for hot ramen needs to be heated now to make.

**Comment by MadmanMartian** [comment]
how hot?
boiling water hot?

**Comment by DrSnips** [comment]
Yeah sounds about right.

**Comment by Blithering** [comment]
It would be a bit of a pain, unless water coolers had the option to dispense hot water, or kettles were added. The noodle pots mention some self heating mechanism anyway on examine.

**Comment by Blithering** [comment]
As for uses, it could increase or decrease the rate at which chemicals are metabolised
Hot milk could make you sleepy, and cold(er) water could wake you up

**Comment by ShiftyRail** [comment]
Fluorine + Hydrogen + Water = Hydrofluoric Acid (aqeuous)
Boil it for a deadly poison on skin contact which attacks clothing
Fluorine + Hydrogen = same thing but it boils at ambiant temperature

**Comment by Shadowmech88** [comment]
What if containers could be placed on the grill to heat the reagents inside them?

**Comment by Blithering** [comment]
Do smart fridges chill things too, or just the fridge closets?

File: code/modules/research/xenoarchaeology/tools/bunsen_burner.dm [hunk 1]
```diff
@@ -49,24 +48,39 @@
...
+		to_chat(user, "<span class='info'>Nothing.</span>")
+	if(held_container)
+		to_chat(user, "<span class='info'>It is holding a:</span>")
+		held_container.examine(user)
+
 /obj/machinery/bunsen_burner/attackby(obj/item/weapon/W, mob/user)
 	if(istype(W, /obj/item/weapon/reagent_containers))
 		var/obj/item/weapon/reagent_containers/R = W
-		if(held_container)
+		if(heating == -1)
```
Comment (PJB3005) [inline code/modules/research/xenoarchaeology/tools/bunsen_burner.dm]: Now you've added `-1` it's becoming a bit of a mess. Use defines to clear it up.

File: code/game/objects/structures/crates_lockers/closets/secure/freezer.dm [hunk 2]
```diff
@@ -3,6 +3,27 @@
 	var/icon_exploded = "fridge_exploded"
 	var/exploded = 0
 
+/obj/structure/closet/secure_closet/freezer/New()
+	..()
+	processing_objects.Add(src)
+
```
Comment (PJB3005) [inline code/game/objects/structures/crates_lockers/closets/secure/freezer.dm:9]: Need a `Destroy()` to remove from processing objects too.

**Comment by MadmanMartian** [comment]
just the fridge closets right now.

Don't want some form of cold-triggered explosive blowing up the chemfridge

_or do we?_

**Review by PJB3005 — CHANGES_REQUESTED** [review]

File: code/game/objects/structures/crates_lockers/closets/secure/freezer.dm (hunk 2 above)
Comment (DamianX) [reply code/game/objects/structures/crates_lockers/closets/secure/freezer.dm:9]: obj/Destroy does it, though.

**Comment by Blithering** [comment]
Smart fridges having cooling would pave the way to unlimited chill pill puns

**Comment by DrSnips** [comment]
A small request for a machine.
Heating pad: uses power from the apc to heat things on it.
-Requires a circuit from research and 2 micro lasers. 
-Better micro lasers increase the max heat or speed of heating. 
-A way to have continuous and reliable heating.
-Moderately low power consumption because it's a smaller pad not a huge oven.
Ideal for xenoarchs because lithium tungstate takes forever to heat or useful for other long to heat chems where the bunsen burner would require several refuels to heat the mixture.

File: code/game/objects/structures/crates_lockers/closets/secure/freezer.dm (hunk 2 above)
Comment (PJB3005) [reply code/game/objects/structures/crates_lockers/closets/secure/freezer.dm:9]: I did not know this.

Comment (MadmanMartian) [reply code/game/objects/structures/crates_lockers/closets/secure/freezer.dm:9]: So, disregard this change request?

File: code/modules/research/xenoarchaeology/tools/bunsen_burner.dm [hunk 3]
```diff
@@ -50,24 +51,39 @@
...
 					var/reagent_transfer = R.reagents.trans_id_to(src, possible_fuel, 10)
 					if(reagent_transfer)
 						to_chat(user, "<span class='notice'>You transfer [reagent_transfer]u of [possible_fuel] from \the [R] to \the [src]</span>")
 						return
-			to_chat(user, "<span class='warning'>You must remove the [held_container] first.</span>")
-			return
 		else
 			if(user.drop_item(W, src))
 				held_container = W
 				to_chat(user, "<span class='notice'>You put the [held_container] onto the [src].</span>")
```
Comment (DrSnips) [inline code/modules/research/xenoarchaeology/tools/bunsen_burner.dm]: \the needed here

File: code/modules/research/xenoarchaeology/tools/bunsen_burner.dm [hunk 4]
```diff
@@ -115,13 +131,18 @@
...
 	icon_state = "bunsen[heating]"
 
+/obj/machinery/bunsen_burner/attack_ghost()
+	return
 
 /obj/machinery/bunsen_burner/attack_hand(mob/user)
 	if(held_container)
-		underlays = null
+		overlays = null
 		to_chat(user, "<span class='notice'>You remove the [held_container] from the [src].</span>")
```
Comment (DrSnips) [inline code/modules/research/xenoarchaeology/tools/bunsen_burner.dm]: \the

**Comment by MadmanMartian** [comment]
@PJB3005

**Review by PJB3005 — APPROVED** [review]

**Comment by PJB3005** [comment]
Well I hate myself right now.

I'll just merge the hotfixes and go to bed I guess.

**Pull request merged**